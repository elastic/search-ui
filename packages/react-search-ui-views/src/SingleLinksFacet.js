import PropTypes from "prop-types";
import React from "react";

import { FacetValue } from "./types";
import { getFilterValueDisplay } from "./view-helpers";
import { appendClassName } from "./view-helpers";

function SingleLinksFacet({ className, label, onRemove, onSelect, options }) {
  const value = options.filter(o => o.selected).map(o => o.value)[0];
  return (
    <div className={appendClassName("sui-facet", className)}>
      <div>
        <div className="sui-facet__title">{label}</div>
        <ul className="sui-single-option-facet">
          {value && (
            <li className="sui-single-option-facet__selected">
              {getFilterValueDisplay(value)}{" "}
              <span className="sui-single-option-facet__remove">
                (
                <a
                  onClick={e => {
                    e.preventDefault();
                    onRemove(value);
                  }}
                  href="/"
                >
                  Remove
                </a>
                )
              </span>
            </li>
          )}
          {!value &&
            options.map(option => (
              <li
                className="sui-single-option-facet__item"
                key={getFilterValueDisplay(option.value)}
              >
                <a
                  className="sui-single-option-facet__link"
                  href="/"
                  onClick={e => {
                    e.preventDefault();
                    onSelect(option.value);
                  }}
                >
                  {getFilterValueDisplay(option.value)}
                </a>{" "}
                <span className="sui-facet__count">{option.count}</span>
              </li>
            ))}
        </ul>
      </div>
    </div>
  );
}

SingleLinksFacet.propTypes = {
  label: PropTypes.string.isRequired,
  onRemove: PropTypes.func.isRequired,
  onSelect: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(FacetValue).isRequired,
  className: PropTypes.string
};

export default SingleLinksFacet;
