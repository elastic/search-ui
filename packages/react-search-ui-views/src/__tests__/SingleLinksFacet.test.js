import React from "react";
import SingleLinksFacet from "../SingleLinksFacet";
import { shallow } from "enzyme";

const params = {
  label: "Facet",
  onRemove: () => {},
  onSelect: () => {},
  options: [
    { value: "1", count: 1, selected: false },
    { value: "2", count: 1, selected: false }
  ]
};

it("renders correctly", () => {
  const wrapper = shallow(<SingleLinksFacet {...params} />);
  expect(wrapper).toMatchSnapshot();
});

it("renders falsey values correctly", () => {
  const wrapper = shallow(
    <SingleLinksFacet
      {...params}
      options={[
        {
          value: 0,
          count: 10,
          selected: false
        },
        {
          value: false,
          count: 20,
          selected: false
        },
        {
          value: "",
          count: 30,
          selected: false
        }
      ]}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

describe("determining selected option", () => {
  it("will correctly determine which of the options is selected", () => {
    const wrapper = shallow(
      <SingleLinksFacet
        {...params}
        options={[
          { value: "1", count: 1, selected: true },
          { value: "2", count: 1, selected: false }
        ]}
      />
    );
    expect(wrapper.find("li").length).toBe(1);
    expect(wrapper.find("li").text()).toBe("1 (Remove)");
  });

  // This shouldn't ever happen, but if it does, it should use the first selected value
  it("will used the first selected option when multiple options are selected", () => {
    const wrapper = shallow(
      <SingleLinksFacet
        {...params}
        options={[
          { value: "1", count: 1, selected: true },
          { value: "2", count: 1, selected: true }
        ]}
      />
    );
    expect(wrapper.find("li").length).toBe(1);
    expect(wrapper.find("li").text()).toBe("1 (Remove)");
  });

  it("will correctly determine when no value is selected", () => {
    const wrapper = shallow(
      <SingleLinksFacet
        {...params}
        options={[
          { value: "1", count: 1, selected: false },
          { value: "2", count: 1, selected: false }
        ]}
      />
    );
    expect(wrapper.find("li").length).toBe(2);
  });
});

it("renders with className prop applied", () => {
  const customClassName = "test-class";
  const wrapper = shallow(
    <SingleLinksFacet {...params} className={customClassName} />
  );
  const { className } = wrapper.props();
  expect(className).toEqual("sui-facet test-class");
});
