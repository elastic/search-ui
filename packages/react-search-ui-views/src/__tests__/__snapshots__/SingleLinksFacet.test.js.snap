// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sui-facet"
>
  <div>
    <div
      className="sui-facet__title"
    >
      Facet
    </div>
    <ul
      className="sui-single-option-facet"
    >
      <li
        className="sui-single-option-facet__item"
        key="1"
      >
        <a
          className="sui-single-option-facet__link"
          href="/"
          onClick={[Function]}
        >
          1
        </a>
         
        <span
          className="sui-facet__count"
        >
          1
        </span>
      </li>
      <li
        className="sui-single-option-facet__item"
        key="2"
      >
        <a
          className="sui-single-option-facet__link"
          href="/"
          onClick={[Function]}
        >
          2
        </a>
         
        <span
          className="sui-facet__count"
        >
          1
        </span>
      </li>
    </ul>
  </div>
</div>
`;
